include/swift/AST/Module.h

```diff
@@ -555,6 +555,18 @@ class FileUnit : public DeclContext {
     return nullptr;
   }
 
+  /// Directly look for a nested type declared within this module inside the
+  /// given nominal type (including any extensions).
+  ///
+  /// This is a fast-path hack to avoid circular dependencies in deserialization
+  /// and the Clang importer.
+  ///
```

Comment:
That's correct. It could probably be non-failable but since the eventual goal is to remove it altogether…no reason not to be cautious.